### ISSUE TITLE
Buttons: allow links that look like buttons to be triggered

### ISSUE THREAD

**Issue by jAigret-Bix** [uswds/uswds#3690]
USAGov accessibility [test results](https://docs.google.com/document/d/1g5beRCFgb5M7xYrEpBPuZy39AKYxtCXzT54FHTPiRgg/edit)

Visual Appearance of Control Does Not Match Function:
The Back button is coded as a link but is styled to look like a button. Visually sighted users see a button, but for screen reader users it is announced as a link. 
Add the ARIA role="button" so that screen readers announce it as a button. Additional JavaScript will also need to be added so that keyboard users can activate the button using both the Enter and Spacebar keys.

![Button](https://user-images.githubusercontent.com/67592482/94596863-c1670b00-025a-11eb-85e2-a1f2e954168c.png)

## Todo
- [x] Allow links that look like buttons to be triggered by spacebar
- [ ] Update guidance on Buttons if needed.

**Comment by mejiaj** [comment]
Does this mean we should add a check w/ JS for all `usa-button`s on the page and check if they're a `<a>` and add the functionality that way?

**Comment by iamjolly** [comment]
I have thoughts on this. Links that look like buttons but take someone to a new page should remain as links (do not add the button role).

**Comment by thisisdano** [comment]
OK, I've stewed on this for quite some time, and I think I finally have an opinion: interactive elements that look like buttons should trigger with the spacebar. This is exactly the point raised in this issue, but I took the long way to get around to an opinion. I got myself pretty bogged down in links-vs-buttons, and while I think I still don't have clear research around what users expect when they find a button, every bit of user testing I even do points to "whatever something appears like, that's what users expect it to be" — if keyboard users expect to trigger button actions with the spacebar, they should not have to think about the content of the button to understand how to interact with it. So,

1) If it looks like a button, the `spacebar` should trigger it.

But _is_ is a button. This still matters and @iamjolly's point stands. Does a link that looks like a button get the role of button? I am inclined to say that it doesn't but our colleagues at gov.uk differ. See [the `Start buttons` documentation](https://design-system.service.gov.uk/components/button/) at the gov.uk design system site. Their system uses buttons _far_ less frequently for links than our does, however. They pretty much only suggest using a button link in a single situation. Our situation is different. 

Screen readers will not know whether a link looks like a link or looks like a button. It _appears_ in the role we assign it. For these users, we should be as clear as possible about what the element is. If it's a link, it should vocalize as a link, if it's a button, it should vocalize as a button. Screen readers have a more direct connection to what the element _is_, our markup should reflect that. So,

2) Markup should reflect what the element is.

@mejiaj Can we develop a solution that supports both points? That supports `spacebar` interaction, but does so without requiring `role="button"`?

@iamjolly Is my reasoning solid? How does this approach sound to you? I am sorry this has been sitting for so long!

**Comment by iamjolly** [comment]
@thisisdano - I like your thinking here. 

Is there a way to provide guidance within USWDS for when to use a link styled as a `button` so that the prevalence of that practice is limited? In most cases, links should remain links with only very specific exceptions. 

I like how Gov.uk makes the exception for a Start button. I agree that if we can allow the button-like link to remain a link (not adding the `role="button"` attribute) while using scripting to add `spacebar` support for those instances, that could be ideal.
